src/flb_record_accessor.c

```diff
@@ -182,7 +182,7 @@ static int ra_parse_buffer(struct flb_record_accessor *ra, flb_sds_t buf)
                 /* ignore '.' if it is inside a string/subkey */
                 continue;
             }
-            else if (buf[end] == '.' || buf[end] == ' ' || buf[end] == ',' || buf[end] == '"') {
```

Comment:
why `buf[end] == '.'` goes away ?

Reply:
It is to allow dot(.) .

I think '.' is already treated as IDENTIFIER on current master.
https://github.com/fluent/fluent-bit/blob/b6668b157b2310cd155f1c1ee5f70b4503f73017/src/record_accessor/ra.l#L56

However this `buf[end] == '.'` doesn't allow dot(.) and break the loop.
Then the buf will be truncated.